docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,90 @@
+= Git Submodule pattern
+
+== How this workflow pattern works:
+
+1. Create a "downstream" repo.
+2. Track your upstream git repo with a git submodule reference.
+3. Onboard to {ProductName}, use link:https://github.com/renovatebot/renovate[Renovate] to keep your Git submodule references up to date.
+4. In your downstream repo, apply patches in your Containerfile if you need to carry downstream patches.
+
+== Implementing this solution pattern
+=== Create a downstream repo
+
+A **downstream repo** is a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+The downstream repo will:
+
+* be used to hold the git submodule reference to the upstream repository.
+* contain a Containerfile that references and pulls in the git submodule(s) content.
+* become a Component of our {ProductName} application.
+
+=== Create the git submodule for the upstream repo to track it
+
+An **upstream repo** is a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+After creating the downstream repo, add a git submodule for the upstream repo:
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repo>
+  url = https://github.com/<namespace>/<project>.git
+  branch = main # Ensure the branch is correct
+----
+See the link:https://git-scm.com/docs/gitsubmodules[Git documentation] for more details on configuring git submodules.
+
+=== Onboard the project onto {ProductName}
+
+After creating the repository, onboard to {ProductName} and configure your application and its components.
+
+**Note:**
+The xref:/getting-started/index.adoc[Getting started docs] are currently geared towards configuring YAML files.
```

Comment:
What do you mean YAML files? i.e. there is no documentation for onboarding via the UI? Instead, the CRs need to be created manually?

I think that that can be left out of _this_ context as it is relevant at the xref target.

Reply:
I removed this part. Most of my experience with Konflux is using the UI. I have yet to do anything via the CLI or define CRs in Konflux.

Reply:
Ah, by YAML files, you just mean defining the CRs in a yaml and creating them with the CLI?

We need to improve the documentation to include both CLI and UI flows. I don't think that we need to make the call-out in this page.

Reply:
Yeah, I'm just gaining experience defining CRs in a yaml and configuring things in the CLI. This section has been removed